nitro-protocol/test/test-helpers.ts

```diff
@@ -335,3 +345,95 @@ export const largeOutcome = (
       ]
     : [];
 };
+
+// Deep comparison
+// Copied from https://stackoverflow.com/a/10316616/10157235
```

Comment:
I'm a little uneasy about committing this code here. 

1. Do we really need this? I think in the past we have relied on[ jest's [`toEqual`](https://jestjs.io/docs/expect#toequalvalue) or [`.toMatchObject`](https://jestjs.io/docs/expect#tomatchobjectobject)
2. If we really need custom deep equality / comparison, is there a library we can import to do that?

Reply:
Somehow, `toEqual` was not working for me properly, but now it does.
I remove `deepEquals` and all logic related to it.